base/system/services/database.c

```diff
@@ -1326,8 +1429,9 @@ ScmControlService(HANDLE hControlPipe,
     DWORD dwError = ERROR_SUCCESS;
     BOOL bResult;
     OVERLAPPED Overlapped = {0};
+    HANDLE hControlPipe = lpService->lpImage->hControlPipe;
 
-    DPRINT("ScmControlService() called\n");
+    DPRINT("ScmControlService(%S, %d) called\n", lpService->lpServiceName, dwControl);
```

Comment:
Note that the lpServiceName is already given via the pServiceName parameter.
Also, the reason why the function took the control pipe separate was in order to take in input only the minimal necessary stuff to operate; it doesn't need the whole SERVICE object.